xapian-core/include/xapian/weight.h

```diff
@@ -1082,6 +1082,80 @@ class XAPIAN_VISIBILITY_DEFAULT PL2Weight : public Weight {
     double get_maxextra() const;
 };
 
+/// Xapian::Weight subclass implementing the PL2+ probabilistic formula.
+class XAPIAN_VISIBILITY_DEFAULT PL2PlusWeight : public PL2Weight {
+
+  public:
+    /** Construct a PL2PlusWeight.
+     *
+     *  @param c  A non-negative and non zero parameter controlling the extent
+     *		  of the normalization of the wdf to the document length. The
+     *		  default value of 1 is suitable for longer queries but it may
+     *		  need to be changed for shorter queries. For more information,
+     *		  please refer to Gianni Amati's PHD thesis titled
+     *		  Probabilistic Models for Information Retrieval based on
+     *		  Divergence from Randomness.
+     *  
+     *  @param delta  A parameter for pseudo tf value to control the scale
+     *                of the tf lower bound. Delta(δ) should be a positive
+     *                real number or in other words, non-negative and non-zero.
+     *                It can be tuned for e.g from 0.1 to 1.5 in increments of
+     *                0.1 or so. Experiments have shown that PL2+ works effectivoly
```

Comment:
As with BM25+, "for e.g." would be better as just "e.g." or "for example" (I'd probably go for the latter myself).

"a positive real number or in other words, non-negative and non-zero" seems needlessly complicated - positive doesn't include zero, so clarification doesn't seem necessary.